outlook/src/taskpane/components/Log/Logger.tsx

```diff
@@ -27,21 +34,141 @@ class Logger extends React.Component<LoggerProps, LoggerState> {
+    private fetchAttachmentContent(attachment, index): Promise<any> {
+        return new Promise<any>((resolve) => {
+            if (attachment.size > elementSizeThreshold * 1024 * 1024) {
+                if (attachment.isInline && attachment.contentType.indexOf("image") >= 0) {
+                    resolve({
+                        name: attachment.name, inline: true,
+                        oversize: true, index
+                    });
+                } else {
+                    resolve({
+                        name: attachment.name, inline: false,
+                        oversize: true, index
+                    });
+                }
+            }
+            Office.context.mailbox.item.getAttachmentContentAsync(attachment.id, asyncResult => {
+                if (attachment.isInline && attachment.contentType.indexOf("image") >= 0) {
+                    resolve({
+                        name: attachment.name,
+                        content: asyncResult.value.content,
+                        inline: true,
+                        index: index,
+                    });
+                } else {
+                    resolve({
+                        name: attachment.name,
+                        content: asyncResult.value.content,
+                        inline: false,
+                        index: index,
+                    });
+                }
+            });
+        });
+    };
+
+    private logRequest = async (event): Promise<any> => {
         event.stopPropagation();
 
         this.setState({logged: 1});
-        Office.context.mailbox.item.body.getAsync(Office.CoercionType.Html, (result) => {
+        Office.context.mailbox.item.body.getAsync(Office.CoercionType.Html, async (result) => {
             const msgHeader = '<div>From: ' + Office.context.mailbox.item.sender.emailAddress + '</div><br/>';
-            const msgFooter = '<br/><div class="text-muted font-italic">Logged from <a href="https://www.odoo.com/documentation/user/crm/optimize/mail_client_extension.html" target="_blank">Outlook Inbox</a></div>';
+            const msgFooter = '<br/><div class="text-muted font-italic">Logged from <a href="https://www.odoo.com/documentation/14.0/applications/sales/crm/optimize/outlook_extension.html" target="_blank">Outlook Inbox</a></div>';
             const body = result.value.split('<div id="x_appendonsend"></div>')[0]; // Remove the history and only log the most recent message.
             const message = msgHeader + body + msgFooter;
 
             const requestJson = {
                 res_id: this.props.resId,
                 model: this.props.model,
-                message: message
+                message: message,
+                attachments: [],
+            }
+
+            const doc = new DOMParser().parseFromString(requestJson.message, "text/html");
+
+            const officeAttachmentDetails = Office.context.mailbox.item.attachments;
+
+            let size = 0;
```

Comment:
You can group some global variables / init at the beginning here, without having so much blank space:

```js
const doc = new DOMParser().parseFromString(requestJson.message, "text/html");
const officeAttachmentDetails = Office.context.mailbox.item.attachments;
let size = 0;
const promises: any[] = [];
```

Would also rename "size" to "totalSize".

Reply:
Okay, init/global variables are now put on top.